src/test/java/uk/gov/nationalarchives/pdi/step/jena/serializer/JenaSerializerStepIT.java

```diff
@@ -0,0 +1,192 @@
+                }});
+
+        // First pass create model, second serialise
+        step.processRow(meta, data);
+        step.processRow(meta, data);
+
+        File f = new File(expectedFilename);
+
+        assertTrue(f.exists() && !f.isDirectory());
+    }
+
+    @Test
+    public void resolves_serialization_format_variable() throws KettleException {
+        final JenaSerializerStepMeta meta = getMeta();
+        final StepDataInterface data = getData();
+        final StepMockHelper helper = mockHelper();
+        final JenaSerializerStep step = mockStep(helper);
+
+        meta.setJenaModelField("model");
+        meta.setFileDetail(new JenaSerializerStepMeta.FileDetail());
+
+        step.setVariable("serialisationFormatVar", "RDF/XML");
+        meta.setSerializationFormat("${serialisationFormatVar}");
+
+        // First pass create model, second serialise
+        step.processRow(meta, data);
+        final boolean rowProcessed = step.processRow(meta, data);
+
+        // Would throw if var was not resolved
+        assertFalse(rowProcessed);
+    }
+
+    private StepDataInterface getData() {
+        return new JenaSerializerStepData() {{
+            init();
+        }};
+    }
+
+    private static JenaSerializerStepMeta getMeta() {
+        return new JenaSerializerStepMeta() {{
+            setDefault();
+        }};
+    }
+
+    private static StepMockHelper mockHelper() {
+        final StepMockHelper helper = new StepMockHelper<>("Serialize Jena Model", JenaSerializerStepMeta.class, JenaSerializerStepData.class);
+
+        when(helper.logChannelInterfaceFactory.create(any(), any(LoggingObjectInterface.class))).thenReturn(helper.logChannelInterface);
+        when(helper.trans.isRunning()).thenReturn(true);
+
+        return helper;
+    }
+
+    private JenaSerializerStep mockStep(StepMockHelper helper) throws KettleException {
+        final JenaSerializerStep step = Mockito.spy(new JenaSerializerStep(helper.stepMeta, helper.stepDataInterface, 0, helper.transMeta, helper.trans));
+
+        step.setRowHandler(
+                new StaticDataRowHandler(
+                        new ArrayList() {{
+                            add(new Object[]{
+                                    ModelFactory.createDefaultModel()});
+                        }}
+                )
+        );
+
+        final RowMeta inputRowSchema = new RowMeta() {{
+            addValueMeta(new ValueMetaInteger("model"));
+        }};
+
+        doReturn(inputRowSchema).when(step).getInputRowMeta();
+
+        return step;
+    }
+
+    private class StaticDataRowHandler implements RowHandler {
+        private final LinkedList<Object[]> rows;
+
+        public StaticDataRowHandler(Collection<Object[]> rows) {
+            this.rows = new LinkedList<>(rows);
+        }
+
+        @Override
+        public Object[] getRow() {
+            if (rows.isEmpty()) {
+                return null;
+            } else {
+                return rows.remove();

```

Comment:
Is this meant to be a `pop`?

Reply:
No. Proper type makes it clearer:
```java
private final Queue<Object[]> rows;
```

Reply:
Okay. In Java if you don't need concurrent access to a queue, I would suggest instead using ArrayDeque and it's interface Deque.